### PR TITLE
#6541: Add optional tensor support for bmm

### PR DESCRIPTION
**Pull request by Aswinmcw**
Add optional output tensor support for batch matmul
All Post commit test : https://github.com/tenstorrent-metal/tt-metal/actions/runs/8342592473

### CODE REVIEW

**Review by eyonland — CHANGES_REQUESTED** [review]
Please add the reference back to the primary ticket in the description.

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_bmm.py [hunk 1]
```diff
@@ -33,8 +33,9 @@ def test_moreh_bmm(shape, device):
 
     # tt bmm
     cpu_layout = ttl.tensor.Layout.ROW_MAJOR
+    tt_output_tensor = None
     tt_out = (
-        ttl.operations.primary.moreh_bmm(tt_input, tt_mat2)
+        ttl.operations.primary.moreh_bmm(tt_input, tt_mat2, tt_output_tensor)
         .cpu()
```
Comment (eyonland) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_bmm.py:39]: Please add a test showing that the optional output tensor does not even need to be provided.  And one that shows that the optional output tensor was provided and returned back by the op.

**Review by eyonland — CHANGES_REQUESTED** [review]
Please add a test showing that the optional output tensor does not even need to be provided.  And one that shows that the optional output tensor was provided and returned back by the op.

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_bmm.py [hunk 2]
```diff
@@ -33,13 +40,29 @@ def test_moreh_bmm(shape, device):
...
+    if optional_tensor == True:
+        torch_output = torch.randint(-2, 3, output_shape, dtype=torch.bfloat16)
+        tt_optional_tensor = (
+            ttl.tensor.Tensor(torch_output, ttl.tensor.DataType.BFLOAT16)
+            .pad_to_tile(1)
+            .to(ttl.tensor.Layout.TILE)
+            .to(device)
+        )
+        (
+            ttl.operations.primary.moreh_bmm(tt_input, tt_mat2, tt_optional_tensor)
```
Comment (eyonland) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_bmm.py:52]: Please add tt_out = ttl.operations.primary.moreh_bmm(tt_input, tt_mat2, tt_optional_tensor)

**Review by eyonland — CHANGES_REQUESTED** [review]
Every test needs to show that the op returned back the optional output tensor it was given.

**Comment by Aswinmcw** [comment]
Closing this PR as Moreh team will implement

**Pull request closed without merging**